var/spack/repos/builtin/packages/otf2/package.py

```diff
@@ -40,5 +40,8 @@ class Otf2(AutotoolsPackage):
     version('1.3.1', 'd0ffc4e858455ace4f596f910e68c9f2')
     version('1.2.1', '8fb3e11fb7489896596ae2c7c83d7fc8')
 
+    def url_for_version(self, version):
+        return 'http://www.vi-hps.org/upload/packages/otf2/otf2-{0}.tar.gz'.format(version.up_to(2))
```

Comment:
This doesn't make sense to me. You're saying that both 1.5.1 and 1.5.2 are downloaded from:
http://www.vi-hps.org/upload/packages/otf2/otf2-1.5.tar.gz

I think the default `url` should be fine. You shouldn't need a `url_for_version` for this package.